src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,174 @@
+     * @return String array of messages to be printed
+     * @throws InvalidIndexException if index is out of bounds
+     */
+    public String[] markTaskDoneAtIndex(int index) throws InvalidIndexException {
+        try {
+            Task task = tasks.get(index);
+            task.markDone();
+            return new String[] {
+                StringConstants.DONE_MESSAGE,
+                "  " + task.toString()
+            };
+        } catch (IndexOutOfBoundsException e) {
+            throw new InvalidIndexException();
+        }
+    }
+
+    /**
+     * Delete task at index.
+     * @param index index of task to delete
+     * @return String array of messages to be printed
+     * @throws InvalidIndexException if index is out of bounds
+     */
+    public String[] deleteTaskAtIndex(int index) throws InvalidIndexException {
+        try {
+            Task task = tasks.get(index);
+            tasks.remove(index);
+            return new String[] {
+                StringConstants.DELETE_MESSAGE,
+                "  " + task.toString(),
+                String.format(StringConstants.COUNT_MESSAGE, tasks.size())
+            };
+        } catch (IndexOutOfBoundsException e) {
+            throw new InvalidIndexException();
+        }
+    }
+
+    /**
+     * Gets task at index.
+     * @param index index of task to get
+     * @return task at index
+     * @throws InvalidIndexException if index is out of bounds
+     */
+    public Task getTaskAtIndex(int index) throws InvalidIndexException {
+        try {
+            return tasks.get(index);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new InvalidIndexException();
+        }
+    }
+
+    /**
+     * Prints all the user tasks.
+     */
+    public void printTaskList() {
+        String[] messages = new String[tasks.size()];
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            messages[i] = String.format("%d.%s", i + 1, task.toString());
+        }
+        Ui.printMessagesBetweenLines(messages);
+    }
+
+    /**
+     * Finds tasks matching with keyword String and prints them.
+     * @param command user command String
+     */
+    public void find(String command) {
+        try {
+            String keyWordString = Parser.getFindKeyWordString(command);
+            ArrayList<String> messages = new ArrayList<>();
+            messages.add(StringConstants.TASK_FOUND_MESSAGE);
+            int i = 1;
+            for (Task task: tasks) {
+                if (task.getName().contains(keyWordString)) {
+                    messages.add(String.format("%d.%s", i, task.toString()));
+                    i++;
+                }
+            }
+            if (messages.size() == 1) {
+                Ui.printMessageBetweenLines(StringConstants.NO_TASK_FOUND_MESSAGE);
+            } else {
+                Ui.printMessagesBetweenLines(messages.toArray(new String[0]));
+            }
+
```

Comment:
extra blank line, a couple of other similar careless mistakes as well, will stop mentioning from here onwards.

Reply:
Prefer this line break for readability